src/components/timeline/Row.svelte

```diff
@@ -128,9 +257,31 @@
   $: hasActivityLayer = !!layers.find(layer => layer.chartType === 'activity');
   $: hasResourceLayer = !!layers.find(layer => layer.chartType === 'line' || layer.chartType === 'x-range');
 
+  // Track resource loading status for this Row
+  $: if (resourceRequestMap) {
+    const newLoadedResources: Resource[] = [];
+    const newLoadingErrors: string[] = [];
+    Object.values(resourceRequestMap).forEach(resourceRequest => {
+      if (resourceRequest.resource) {
+        newLoadedResources.push(resourceRequest.resource);
+      }
+      if (resourceRequest.error) {
+        newLoadingErrors.push(resourceRequest.error);
+      }
+    });
+    loadedResources = newLoadedResources;
+    loadingErrors = newLoadingErrors;
+
+    // Consider row to be loading if the number of completed resource requests (loaded or error state)
+    // is not equal to the total number of resource requests
+    anyResourcesLoading = loadedResources.length + loadingErrors.length !== Object.keys(resourceRequestMap).length;
+  }
+
   // Compute scale domains for axes since it is optionally defined in the view
-  $: if ($allResources && yAxes) {
-    yAxesWithScaleDomains = getYAxesWithScaleDomains(yAxes, layers, resourcesByViewLayerId, viewTimeRange);
+  $: if (loadedResources && yAxes) {
+    yAxesWithScaleDomains = getYAxesWithScaleDomains(yAxes, layers, loadedResources, viewTimeRange);
+    // In this case we can directly mutate the cache since we don't need to react to it.
+    $yAxesWithScaleDomainsCache[id] = yAxesWithScaleDomains;
```

Comment:
Dispatch a "yAxisChange" event so that Timeline or TimelinePanel could be the one responsible for managing the store update.